python/pyspark/statcounter.py

```diff
@@ -54,12 +59,13 @@ def merge(self, value):
         return self
 
     # Merge another StatCounter into this one, adding up the internal statistics.
-    def mergeStats(self, other):
+    def mergeStats(self, other: "StatCounter") -> "StatCounter":
         if not isinstance(other, StatCounter):
             raise TypeError("Can only merge StatCounter but got %s" % type(other))
 
         if other is self:  # reference equality holds
-            self.merge(copy.deepcopy(other))  # Avoid overwriting fields in a weird order
+            self.merge(
+                copy.deepcopy(cast(float, other)))  # Avoid overwriting fields in a weird order
```

Comment:
this looks weird. when other is self, shouldn't `other` be `StatCounter`?

Reply:
updated. Thanks!